ktlint/src/main/resources/ktlint-git-pre-commit-hook.sh

```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 # https://github.com/pinterest/ktlint pre-commit hook
-git diff --name-only --cached --relative | grep '\.kt[s"]\?$' | xargs ktlint --relative
+git diff --name-only --cached --relative | grep '\.kt[s"]\?$' | xargs --no-run-if-empty ktlint --relative
```

Comment:
Cool change. I was not aware of that xargs option. Same needs to be done for the other hooks to keep it all consistent.

Reply:
Unfortunately, this won't work for example on MacOs. The "flag --no-run-if-empty" is not recognized (see https://stackoverflow.com/questions/8296710/how-to-ignore-xargs-commands-if-stdin-input-is-empty) as balnk inut is ignored by default. The change therefore can not be merged as it is right now.

Reply:
In order to keep things as 'sh'-oriented as possible (i.e. no bash embellishments) may I suggest something like:
```suggestion
[ `uname -s` != "Darwin" ] && no_run=--no-run-if-empty
git diff --name-only --cached --relative | grep '\.kt[s"]\?$' | xargs $no_run ktlint --relative
```

Or another option which would cater for other older/bsd-style `xargs` would be:
```suggestion
`xargs -r </dev/null >/dev/null 2>&1` && no_run=--no-run-if-empty
git diff --name-only --cached --relative | grep '\.kt[s"]\?$' | xargs $no_run ktlint --relative
```
but this seems really verbose and possibly even over-kill.

Just a thought.

Reply:
I am not sure whether it will work on all OS but at least it does work on MacOs ;-)

I would prefer the first alternative as it is more readable but I would like a clarifying comment above the uname command to explain that this is MacOS specific.

The same trick should be applied to all other hooks as well. Please search on "xargs" to find them all.

Reply:
First suggestion looks best IMO too.

Just tweaked it to adhere to ShellCheck